File: backend/src/main/java/com/yourwatchrental/watchrental/rental/dto/response/RentalResponseDTO.java
package com.yourwatchrental.watchrental.rental.dto.response;

import com.yourwatchrental.watchrental.branch.dto.BranchResponseDTO;
import com.yourwatchrental.watchrental.rental.PaymentMethod;
import com.yourwatchrental.watchrental.rental.PaymentStatus;
import com.yourwatchrental.watchrental.rental.RentalStatus;
import com.yourwatchrental.watchrental.user.dto.response.UserResponseDTO;
import com.yourwatchrental.watchrental.watch.dto.response.WatchFullInfoResponseDTO;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

public record RentalResponseDTO(

        UUID id,

        LocalDate startDate,

        LocalDate endDate,

        BigDecimal totalPrice,

        PaymentMethod paymentMethod,

        RentalStatus rentalStatus,

        PaymentStatus paymentStatus,

        BranchResponseDTO branch,

        UserResponseDTO user,

        WatchFullInfoResponseDTO watch

) {
}

File: frontend/src/app/core/models/rentals/rental-response.dto.ts
import { BranchResponseDTO } from "../branch/branch-response.dto"
import { UserResponseDTO } from "../profile/user-response.dto"
import { WatchFullInfoResponseDTO } from "../watches/watch-full-info-response.dto"
import { PaymentMethod } from "./payment-method"
import { PaymentStatus } from "./payment-status"
import { RentalStatus } from "./rental-status"

export interface RentalResponseDTO{
        id: string;
        startDate: string;
        endDate: string;
        totalPrice: string;
        paymentMethod: PaymentMethod;
        rentalStatus: RentalStatus;
        paymentStatus: PaymentStatus;
        branch: BranchResponseDTO;
        user: UserResponseDTO;
        watch: WatchFullInfoResponseDTO;
        createdAt: string
}